src/components/MainNav/MainNav.js

```diff
@@ -19,206 +14,107 @@ import {
-
+import { logout } from '../../loginServices';
 import css from './MainNav.css';
 import NavButton from './NavButton';
 import NavDivider from './NavDivider';
 import { CurrentAppGroup } from './CurrentApp';
 import ProfileDropdown from './ProfileDropdown';
 import AppList from './AppList';
 import { SkipLink } from './components';
-import { packageName } from '../../constants';
 
-import settingsIcon from './settings.svg';
+import { useAppOrderContext } from './AppOrderProvider';
 
-class MainNav extends Component {
-  static propTypes = {
-    intl: PropTypes.object,
-    stripes: PropTypes.shape({
-      config: PropTypes.shape({
-        showPerms: PropTypes.bool,
-        helpUrl: PropTypes.string,
-      }),
-      store: PropTypes.shape({
-        dispatch: PropTypes.func.isRequired,
-      }),
-      hasPerm: PropTypes.func.isRequired,
-      withOkapi: PropTypes.bool,
-    }),
-    history: PropTypes.shape({
-      listen: PropTypes.func.isRequired,
-      replace: PropTypes.func.isRequired,
-      push: PropTypes.func.isRequired,
-    }).isRequired,
-    location: PropTypes.shape({
-      pathname: PropTypes.string,
-    }).isRequired,
-    modules: PropTypes.shape({
-      app: PropTypes.arrayOf(PropTypes.object),
-    }),
-    queryClient: PropTypes.object.isRequired,
-  };
+import { useStripes } from '../../StripesContext';
+import { useModules } from '../../ModulesContext';
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      userMenuOpen: false,
-    };
-    this.store = props.stripes.store;
-    this.getAppList = this.getAppList.bind(this);
-  }
+const MainNav = () => {
+  const {
+    apps,
+  } = useAppOrderContext();
+  const queryClient = useQueryClient();
+  const stripes = useStripes();
+  const location = useLocation();
+  const modules = useModules();
+  const history = useHistory();
+  const intl = useIntl();
+
+  const curModule = useRef(getCurrentModule(modules, location));
+  const selectedApp = useRef(apps.find(entry => entry.active)).current;
+  const helpUrl = useRef(stripes.config.helpUrl ?? 'https://docs.folio.org').current;
 
-  componentDidMount() {
-    let curQuery = getLocationQuery(this.props.location);
+  useEffect(() => {
+    let curQuery = getLocationQuery(location);
     const prevQueryState = {};
 
-    this._unsubscribe = this.store.subscribe(() => {
-      const { history, location } = this.props;
-      const module = this.curModule;
-      const state = this.store.getState();
+    const { store } = stripes;
+    const _unsubscribe = store.subscribe(() => {
+      const module = curModule;
+      const state = store.getState();
 
       // If user has timed out, force them to log in again.
       if (state?.okapi?.token && state.okapi.authFailure
         && find(state.okapi.authFailure, { type: 'error', code: 'user.timeout' })) {
-        this.returnToLogin();
+        // this.returnToLogin();
+        logout(state.okapi.url, store, queryClient);
       }
```

Comment:
In RTR-land where tokens only exist in HTTPOnly cookies, we don't have to worry about this condition because `state.okapi.token` will never evaluate to true. IOW, I think the right clean-up here is "remove this entire conditional". 

It looks like the bad commit is `a9b860dfe591f7e7b236640fc1ceb6d25401b30c`, itself a "clean-up" commit  from mega-merge PR #1532. The comment there alludes to fixing up logout, but unfortunately doesn't offer any details about why, specifically, `returnToLogin` was removed. If anybody knows this @zburke character we should find him and tell him to write better commit messages.